Lib/asyncio/tasks.py

```diff
@@ -496,6 +496,11 @@ async def wait_for(fut, timeout, *, loop=None):
             # after wait_for() returns.
             # See https://bugs.python.org/issue32751
             await _cancel_and_wait(fut, loop=loop)
+            # In case task cancellation failed with some
+            # exception, we should re-raise it
+            # See https://bugs.python.org/issue40607
+            if not fut.cancelled() and fut.exception() is not None:
+                raise fut.exception()
```

Comment:
Could it be
```python3
if not fut.cancelled():
    return fut.result()
```
?

Reply:
Building on this, how about--

```python
try:
    return fut.result()
except CancelledError as exc:
    raise TimeoutError() from exc
```

What's nice about this is that you'll see the `CancelledError` that caused the `TimeoutError` in the traceback. This will be even better when this PR is merged: https://github.com/python/cpython/pull/19951
In the traceback, you'll get to see exactly what line of code was interrupted by the cancellation.

Reply:
OK, thanks, I find this even more appropriate way to handle such situations.